test/cycles/basis.jl

```diff
@@ -0,0 +1,69 @@
+@testset "Cycle Basis" begin
+
+    function evaluate(x,y)
+        @test length(x) == length(y)
+        for i=1:length(x)
+            @test isempty(setdiff(x[i],y[i]))
+            @test isempty(setdiff(x[i],y[i]))
+        end
+    end
+
+    # No Edges
+    ex = Graph(1)
+    expected_cyclebasis = Array{Int64,1}[]
+    for g in testgraphs(ex)
+        ex_cyclebasis = cycle_basis(g)
+        @test isempty(ex_cyclebasis)
+    end
+
+    # Only one self-edge
+    ex = Graph(1)
+    add_edge!(ex, 1,1)
+    expected_cyclebasis = Array{Int64,1}[[1]]
+    for g in testgraphs(ex)
+        ex_cyclebasis = cycle_basis(g)
+        evaluate(ex_cyclebasis,expected_cyclebasis)
+    end
+
+    # Graph with one cycle
+    ex = Graph(5)
+    edgs = [(1,2),(2,3),(3,4),(4,1),(1,5)]
+    for e in edgs
+        add_edge!(ex, e)
+    end
+    expected_cyclebasis = Array{Int64,1}[
+        [1,2,3,4] ]
+    for g in testgraphs(ex)
+        ex_cyclebasis = cycle_basis(g)
+        evaluate(ex_cyclebasis,expected_cyclebasis)
+    end    
+
+    # Graph with 2 of 3 cycles forming a basis
+    ex = Graph(4)
+    edgs = [(1,2),(1,3),(2,3),(2,4),(3,4)]
+    for e in edgs
+        add_edge!(ex, e)
+    end
```

Comment:
`ex = Graph(SimpleEdge.(edgs))`?

Reply:
```
using LightGraphs
using LightGraphs.SimpleGraphs
elist = [(1,2),(1,3),(2,3),(2,4),(3,4)]
ex = Graph(SimpleEdge.(elist))
```
This gives me `MethodError: Cannot `convert` an object of type Array{LightGraphs.SimpleGraphs.SimpleEdge{Int64},1} to an object of type LightGraphs.SimpleGraphs.SimpleGraph`. I am probably making a stupid mistake here.

Reply:
It does not look like this constructor exists on the latest stable version, but it does on master:
https://github.com/JuliaGraphs/LightGraphs.jl/blob/master/src/SimpleGraphs/simplegraph.jl
line 96.
It's a bit of a mess to use because the linear algebra lib is failing, but consider the line Seth gave you above works

Reply:
Ok, I changed it and hope the tests pass.

Reply:
Hah! Me too. I can't test on 0.7 at work right now.